### PR TITLE
merge: optimize the countVowels algo

### PR DESCRIPTION
**Pull request by fahimfaisaal**
# Welcome to JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

* [x] simplify the algo by using regex and String.prototype.match method
* [x] modified the JS Doc

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.

### CODE REVIEW

File: String/CountVowels.js [hunk 1]
```diff
@@ -1,25 +1,21 @@
 /**
  * @function countVowels
  * @description Given a string of words or phrases, count the number of vowels.
- * @param {String} url - The input string
- * @return {Number} count
+ * @param {String} str - The input string
+ * @return {Number} - The number of vowel
```
Comment (raklaptudirm) [inline String/CountVowels.js:5]:
```suggestion
 * @return {Number} - The number of vowels
```

File: String/CountVowels.js [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
+ * @param {String} str - The input string
+ * @return {Number} - The number of vowel 
  * @example countVowels("ABCDE") => 2
  * @example countVowels("Hello") => 2
  */
 
 const countVowels = (str) => {
   if (typeof str !== 'string') {
-    throw new TypeError('Input should be a string')
+    throw new TypeError('Input should be a string');
```
Comment (raklaptudirm) [inline String/CountVowels.js:12]:
```suggestion
    throw new TypeError('Input should be a string')
```

File: String/CountVowels.js [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-  return count
+
+  const vowelRegex = /[aeiou]/gi;
+  const vowelsArray = str.match(vowelRegex) || [];
+
+  return vowelsArray.length;
 }
 
-export { countVowels }
+export { countVowels };
```
Comment (raklaptudirm) [inline String/CountVowels.js:21]:
```suggestion
export { countVowels }
```

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

**Comment by fahimfaisaal** [comment]
Mistakenly I have pushed the last two commits, that's why I revert that. I'll add again this in the next pull request.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Format your code with `standard.js`.

**Comment by fahimfaisaal** [comment]
Done ✅

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**